test/repository_test.rb

```diff
@@ -115,6 +115,28 @@
           UserRepository.create(user1)
           user1.id.must_equal id
         end
+
+        describe 'when entity is not persisted' do
+          before do
+            @unpersisted_user = User.new(name: 'M', age: '23')
+          end
+          it 'assigns and persists created_at attribute' do
+            result = UserRepository.create(@unpersisted_user)
+            result.created_at.must_equal result.created_at
+          end
+        end
+
+        describe 'when entity is already persisted' do
+          before do
+            user              = User.new(name: 'D', age: '25')
+            @user             = UserRepository.create(user)
+            @persisted_user   = UserRepository.persist(@user)
```

Comment:
1) I think you misunderstood the `.persist` and `.create`. Please read the method implementation again.

2) You are testing `.create`, is there any reason why you insist on testing `.persist` now?

Below is how it is supposed to be done:

```ruby
before do
  @persisted_user = UserRepository.persist(User.new(name: 'D', age: '25'))
  @created_at = @persisted_user.created_at
end

after do
  UserRepository.delete(@persisted_user)
end

it 'does not touch created_at' do
  UserRepository.create(@persisted_user)
  @persisted_user.created_at.must_equal @created_at
end
```